src/main/java/org/opensearch/knn/index/query/parser/RescoreParser.java

```diff
@@ -106,6 +107,7 @@ public static void streamOutput(StreamOutput out, RescoreContext rescoreContext)
             return;
         }
         out.writeOptionalFloat(rescoreContext == null ? null : rescoreContext.getOversampleFactor());
+        out.writeOptionalBoolean(rescoreContext == null ? null : rescoreContext.isUserProvided());
```

Comment:
Why do we need this? This could lead to a break in BWC. Could we just ensure the optional float encapsulates this info?